### PR TITLE
Add support for Catalog Snapshots to Serverless Operator

### PR DESCRIPTION
**Pull request by wkulhanek**
##### SUMMARY
Add support for Operator Catalog Mirrors to the Serverless Operator. Also download tkn from the installed version of Serverless which always matches the installed version.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
ocp4_workload_serverless

### CODE REVIEW

File: ansible/roles_ocp_workloads/ocp4_workload_serverless/defaults/main.yml [hunk 1]
```diff
@@ -3,25 +3,48 @@ become_override: False
...
-# Version of kn to be installed on the bastion
-ocp4_workload_serverless_kn_version: "0.13.2"
+# Defaults values below are for OpenShift Serverless v1.7.2 (GA)
 
 # Channel to use for the OpenShift Serverless subscription
 ocp4_workload_serverless_channel: "4.4"
 
-# Set InstallPlan approvel to Automatic? If no it is also suggested to set the starting_csv
-# to pin a specific version
+# Set InstallPlan approvel to Automatic? If no it is also suggested
```
Comment (stencell) [inline ansible/roles_ocp_workloads/ocp4_workload_serverless/defaults/main.yml:11]: approval

Also this comment reads funny. Maybe it is the ? after automatice.

File: ansible/roles_ocp_workloads/ocp4_workload_serverless/tasks/remove_workload.yml [hunk 2]
```diff
@@ -114,15 +114,13 @@
     name: "{{ _ocp4_workload_serverless_install_plan_name }}"
     namespace: openshift-operators
 
-- name: Remove remaining knativeservings CRDs
+- name: Remove CatalogSource for mirrored catalog
```
Comment (stencell) [inline ansible/roles_ocp_workloads/ocp4_workload_serverless/tasks/remove_workload.yml:117]: Should there be a conditional on this task? We don't really need to remove this catalog source unless it was deployed?

File: ansible/roles_ocp_workloads/ocp4_workload_serverless/tasks/workload.yml [hunk 3]
```diff
@@ -4,54 +4,51 @@
...
-  become: true
-  command: /usr/bin/kn completion bash >/etc/bash_completion.d/kn
-  args: 
-    creates: /etc/bash_completion.d/kn
+- name: Create Catalogsource for use with catalog mirror
+  when: ocp4_workload_serverless_use_catalog_mirror | bool
+  k8s:
+    state: present
+    definition: "{{ lookup('template', item ) | from_yaml }}"
+  loop:
```
Comment (stencell) [inline ansible/roles_ocp_workloads/ocp4_workload_serverless/tasks/workload.yml:12]: If we're not looping, can we not loop?

File: ansible/roles_ocp_workloads/ocp4_workload_serverless/defaults/main.yml (hunk 1 above)
Comment (wkulhanek) [reply ansible/roles_ocp_workloads/ocp4_workload_serverless/defaults/main.yml:11]: OK. Change made. Maybe the update reads better.

File: ansible/roles_ocp_workloads/ocp4_workload_serverless/tasks/remove_workload.yml (hunk 2 above)
Comment (wkulhanek) [reply ansible/roles_ocp_workloads/ocp4_workload_serverless/tasks/remove_workload.yml:117]: But this is a idempotent statement. If it wasn't there then `state: absent` will already be true.

File: ansible/roles_ocp_workloads/ocp4_workload_serverless/tasks/workload.yml (hunk 3 above)
Comment (wkulhanek) [reply ansible/roles_ocp_workloads/ocp4_workload_serverless/tasks/workload.yml:12]: Updated.

**Pull request merged**